WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueSocialFragment.java

```diff
@@ -0,0 +1,364 @@
+        } else if (mSignupEpilogueListener != null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UPDATE_USERNAME_SUCCEEDED);
+            mSignupEpilogueListener.onContinue();
+        }
+    }
+
+    protected boolean changedDisplayName() {
+        return !StringUtils.equals(getArguments().getString(ARG_DISPLAY_NAME), mDisplayName);
+    }
+
+    protected boolean changedUsername() {
+        return !StringUtils.equals(getArguments().getString(ARG_USERNAME), mUsername);
+    }
+
+    protected void showErrorDialog(String message) {
+        DialogInterface.OnClickListener dialogListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                switch (which) {
+                    case DialogInterface.BUTTON_NEGATIVE:
+                        undoChanges();
+                        break;
+                    case DialogInterface.BUTTON_POSITIVE:
+                        updateAccountOrContinue();
+                        break;
+                }
+            }
+        };
+
+        AlertDialog dialog = new AlertDialog.Builder(new ContextThemeWrapper(getActivity(), R.style.LoginTheme))
+                .setMessage(message)
+                .setNeutralButton(R.string.login_error_button, dialogListener)
+                .setNegativeButton(R.string.signup_epilogue_error_button_negative, dialogListener)
+                .setPositiveButton(R.string.signup_epilogue_error_button_positive, dialogListener)
+                .create();
+        dialog.show();
+    }
+
+    protected void undoChanges() {
+        mDisplayName = getArguments().getString(ARG_DISPLAY_NAME);
+        mInputDisplayName.getEditText().setText(mDisplayName);
+        mUsername = getArguments().getString(ARG_USERNAME);
+        mInputUsername.getEditText().setText(mUsername);
+        updateAccountOrContinue();
+    }
+
+    protected void updateAccountOrContinue() {
+        if (changedDisplayName()) {
+            startProgress();
+            PushAccountSettingsPayload payload = new PushAccountSettingsPayload();
+            payload.params = new HashMap<>();
+            payload.params.put("display_name", mDisplayName);
+            mDispatcher.dispatch(AccountActionBuilder.newPushSettingsAction(payload));
+        } else if (changedUsername()) {
+            startProgress();
+            PushUsernamePayload payload = new PushUsernamePayload(mUsername,
+                    AccountUsernameActionType.KEEP_OLD_SITE_AND_ADDRESS);
+            mDispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload));
+        } else if (mSignupEpilogueListener != null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UNCHANGED);
+            mSignupEpilogueListener.onContinue();
+        }
+    }
+
+    private class DownloadAvatarAndUploadGravatarTask extends AsyncTask<String, Void, Void> {
```

Comment:
Since there is no feedback returned to the UI, can we demote the `AsyncTask` to a simple `Thread` perhaps?

Reply:
I updated the class to a `Thread` in 12ccfd14614e1669a455ca2dafd4f09e41f9baa8.